src/librbd/ImageWatcher.cc

```diff
@@ -728,7 +733,15 @@ Context *ImageWatcher<I>::prepare_quiesce_request(
       delete it->second.first;
       it->second.first = ack_ctx;
     } else {
-      m_task_finisher->queue(new C_ResponseMessage(ack_ctx));
+      int r = 0;
+      auto it = m_async_complete.find(request);
+      if (it != m_async_complete.end()) {
+        r = it->second;
+      } else {
+        // quiesce is already complete but unqiesce not
```

Comment:
Question: I thought `m_async_complete` was tracking the old return code (for 10 minutes). So how does a resent quiesce message fall through the first check in order to need this second level of status code storage?

Reply:
Note, we move a request to complete only after unquiesce is complete (see `unquiesce_ctx` below).

Initially it was this way to track duplicate notifications. After introducing `m_async_complete` it seems we could do the way you describe, i.e. move to complete just after quiesce is complete and track duplicates here. And I was considering this but did not like it very much, becausein this case we rely on `m_async_complete` expiration time (currently 10 minutes) is always bigger than  "quiesce-unquiesce" timeperiod. And although it is hardly likely we ever have a situation where it is not true, still it does not look very nice to me. (E.g. in future the expiration time might become a configurable and someone will set it to very small value).

Still, if you like this approach more, I don't mind to update the code.

Reply:
> in this case we rely on `m_async_complete` expiration time (currently 10 minutes) is always bigger than "quiesce-unquiesce" timeperiod. And although it is hardly likely we ever have a situation where it is not true, still it does not look very nice to me. (E.g. in future the expiration time might become a configurable and someone will set it to very small value).

Could we reset the expiration time each time we receive a duplicate request? 
 
> Still, if you like this approach more, I don't mind to update the code.

I was just trying to avoid the need to add even more code for this edge condition.

Reply:
ok